docs/tutorials/multimodal/advanced_topics/tensorrt.ipynb

```diff
@@ -0,0 +1,428 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2f447fc8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.testing.assert_allclose(y_pred, y_pred_trt, atol=0.01)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a76e541",
+   "metadata": {},
+   "source": [
+    "### Visualize Inference Speed\n",
+    "\n",
+    "We can calculate inference time by dividing the prediction time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6ea1e0e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "infer_speed = batch_size/np.mean(pred_time)\n",
+    "infer_speed_trt = batch_size/np.mean(pred_time_trt)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb5d4255",
+   "metadata": {},
+   "source": [
+    "Then, visualize speed improvements."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9261e156",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "fig, ax = plt.subplots()\n",
+    "fig.set_figheight(1.5)\n",
+    "ax.barh([\"PyTorch\", \"TensorRT\"], [infer_speed, infer_speed_trt])\n",
+    "ax.annotate(f\"{infer_speed:.1f} rows/s\", xy=(infer_speed, 0))\n",
+    "ax.annotate(f\"{infer_speed_trt:.1f} rows/s\", xy=(infer_speed_trt, 1))\n",
+    "_ = plt.xlabel('Inference Speed (rows per second)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "433862ef",
+   "metadata": {},
+   "source": [
+    "We can expect about 2x speedup from `optimize_for_inference()`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "952bb633",
+   "metadata": {},
+   "source": [
+    "### Compare Evaluation Metric\n",
+    "Now that we can achieve better inference speed with `optimize_for_inference()`, but is there any impact to the underlining accuracy loss?\n",
+    "\n",
+    "Let's start with whole test dataset evaluation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eb8ccfee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "metric = predictor.evaluate(test_data)\n",
+    "metric_trt = trt_predictor.evaluate(test_data)\n",
+    "clear_output()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7633720a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "metric_df = pd.DataFrame.from_dict({\"PyTorch\": metric, \"TensorRT\": metric_trt})\n",
+    "metric_df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae972756",
+   "metadata": {},
+   "source": [
+    "The evaluation results are expected to be very close.\n",
+    "\n",
+    "In case there is any significant gap between the evaluation results, try disabling mixed precision by using CUDA execution provider:\n",
+    "\n",
+    "```python\n",
+    "predictor.optimize_for_inference(providers=[\"CUDAExecutionProvider\"])\n",
```

Comment:
`CUDAExecutionProvider ` doesn't use TensorRT. Is there a speedup if using `CUDAExecutionProvider`?

Reply:
Yes, there is, but not as significant as TensorRT counterpart.